apps/nextra/pages/en/build/create-aptos-dapp/templates/custom-indexer.mdx

```diff
@@ -0,0 +1,219 @@
+title: "Create Aptos Dapp Custom Indexer Template"
+---
+
+import { Callout, Steps } from 'nextra/components';
+
+# Create Aptos Dapp Custom Indexer Template
+
+The Custom Indexer template provides a starter dapp with all necessary infrastructure to build a full stack app with custom indexer support.
+
+The Custom Indexer template provides:
+
+- Folder structure - A pre-made dapp folder structure with `src` for frontend, `contract` for Move contract and `indexer` for custom indexer.
+- Dapp infrastructure - All required dependencies a dapp needs to start building on the Aptos network.
+- Wallet Info implementation - Pre-made `WalletInfo` components to demonstrate how one can use to read a connected Wallet info.
+- Message board functionality implementation - Pre-made `MessageBoard` component to create, update and read messages from the Move smart contract.
+- Analytics dashboard - Pre-made `Analytics` component to show the number of messages created and updated.
+- Point program - Minimal example to show you how to define a point program based on events (e.g. create message, update message) and show that on the analytics dashboard, with sorting support.
+
+## Generate the Boilerplate template
+
+On your terminal, navigate to the directory you want to work in and run:
+```bash
+npx create-aptos-dapp@latest
+```
+
+Follow the CLI prompts.
+
+## Getting started
+
+<Steps>
+
+### Publish the contract
+
+Run the below command to publish the contract on-chain:
+```bash
+npm run move:publish
+```
+This command will:
+1. Publish the contract to chain.
+2. Setting the `MODULE_ADDRESS` in the `.env` file to set the contract object address.
+
+### Create a postgres database
+
+Sign up for [Neon postgres](https://neon.tech/) which is our cloud Postgres provider and create a new project. Find the connection string and put it in `DATABASE_URL` in the `.env` file of frontend. 
+
+### Setup the database
+
+Everything here should be run in the `indexer` folder.
+
+Install diesel cli to run migrations.
+
+```sh
+cargo install diesel_cli --no-default-features --features postgres
+```
+
+Run all pending database migrations. This will create all the tables in the database.
+
+```sh
+diesel migration run \
+    --config-file="src/db_migrations/diesel.toml" \
+    --database-url="postgresql://username:password@neon_host/db_name?sslmode=require"
+```
+
+In case you want to revert all migrations to delete all tables because you want to re-index all data.
+
+```sh
+diesel migration revert \
+	--all \
+	--config-file="src/db_migrations/diesel.toml" \
+    --database-url="postgresql://username:password@neon_host/db_name?sslmode=require"
+```
+
+In case you want to make any change to the database, you can add a new migration.
+
+```sh
+diesel migration generate create-abc-table \
+    --config-file="src/db_migrations/diesel.toml"
+```
+
+### Sign up for Aptos Build
+
+Sign up for [Aptos Build](https://developers.aptoslabs.com/), create a new project and get the API token.
```

Comment:
1. is this required? I think it should be optional as there is a free tier for node/faucet/indexer requests and they should use an API key only when it is needed
2. Looks like we dont do anything with this API key

Reply:
gotcha. lets add it here then